internal/cmd/controller/gitops/reconciler/gitjob_controller.go

```diff
@@ -172,37 +195,18 @@ func (r *GitJobReconciler) updateStatus(ctx context.Context, gitRepo *v1alpha1.G
 		}
 
 		currentGitRepo.Status.GitJobStatus = result.Status.String()
+		currentGitRepo.Status.ObservedGeneration = gitRepo.Generation
 
 		for _, con := range result.Conditions {
 			condition.Cond(con.Type.String()).SetStatus(currentGitRepo, string(con.Status))
 			condition.Cond(con.Type.String()).SetMessageIfBlank(currentGitRepo, con.Message)
 			condition.Cond(con.Type.String()).Reason(currentGitRepo, con.Reason)
 		}
 
-		if result.Status == status.FailedStatus {
```

Comment:
Now that we compute the termination message outside `RetryOnConflict`, that termination message may be outdated by the time the status is updated. The PR description suggests that this is intentional, but why? :thinking:

Reply:
Mainly for consistency. It's the job we pass into `updateStatus`. It's the one we computed the status for. I don't think the reconciler should switch "inputs" only because it hits a conflict with other controller. If we want to do that, we need to repeat all the work (status computation), not just parts (termination message) of it. 

That's alsow the reason for setting the generation to the value of the gitrepo that triggered the reconcile.
Gitrepo reconciler might get triggered for that change, too, so it will be consistent eventually.

Reply:
I see, that makes sense to me. Thanks for clarifying :)